File: backend/schemas/checkout.py
# === schemas/checkout.py ===

from typing import List, Optional

from pydantic import BaseModel, Field


class CheckoutBookItem(BaseModel):
    """Represents a single book entry in the checkout summary."""

    isbn: str
    title: str
    authors: str
    quantity: int


class CheckoutSummaryResponse(BaseModel):
    """Response schema for checkout summary with user and book details."""

    user_id: str
    first_name: str
    last_name: str
    email: str
    total_items: int
    distinct_titles: int
    books: List[CheckoutBookItem]


class LockerResponse(BaseModel):
    """Represents an available locker with optional distance information."""

    id: str
    locker_code: str
    street: str
    city: str
    postal_code: str
    lat: float
    lon: float
    distance_km: Optional[float] = None


class CheckoutSubmitRequest(BaseModel):
    """Request schema for submitting a checkout with the chosen locker."""

    locker_id: str = Field(..., description="Chosen locker for delivery")


class CheckoutSubmitResponse(BaseModel):
    """Response schema returned after a successful checkout submission."""

    order_id: str
    shipment_id: str
    pickup_code: str
    locker_code: str
    city: str
    postal_code: str
    message: str

File: frontend/src/app/models/checkout.ts
export interface CheckoutBookItem {
  isbn: string;
  title: string;
  authors: string;
  quantity: number;
}

export interface CheckoutSummaryResponse {
  user_id: string;
  first_name: string;
  last_name: string;
  email: string;
  total_items: number;
  distinct_titles: number;
  books: CheckoutBookItem[];
}

export interface LockerResponse {
  id: string;
  locker_code: string;
  street: string;
  city: string;
  postal_code: string;
  lat: number;
  lon: number;
  distance_km?: number | null;
}

export interface CheckoutSubmitResponse {
  order_id: string;
  shipment_id: string;
  pickup_code: string;
  locker_code: string;
  city: string;
  postal_code: string;
  message: string;
}
